packages/flutter/lib/src/widgets/text_editing_intents.dart

```diff
@@ -115,6 +115,38 @@ class ExtendSelectionByCharacterIntent extends DirectionalCaretMovementIntent {
     : super(forward, collapseSelection);
 }
 
+/// Extends, or moves the current selection from the current
+/// [TextSelection.extent] position to the left or the right character
+/// boundary.
+class HorizontalExtendSelectionByCharacterIntent extends ExtendSelectionByCharacterIntent {
+  /// Creates an [HorizontalExtendSelectionByCharacterIntent].
+  const HorizontalExtendSelectionByCharacterIntent({
+    required this.right,
+    required super.collapseSelection,
+    TextDirection textDirection = TextDirection.ltr,
```

Comment:
The creator of the intent is likely going to have no information about who the receiving text field is going to be, so how does it know what text direction to use?

Reply:
> this field is assigned a value once the direction is specified

I think that's a bit problematic as the widget that emits the intent may not know who the receiving text field is going to be (as a result it has not way to determine the current writing direction since different text fields can have different writing directions). IOW, when you're constructing the intent, there's no way to know determine the correct `TextDirection`.